cvc5_z3py_compat/z3.py

```diff
@@ -5199,6 +5198,25 @@ def set(self, **kwargs):
             _assert(isinstance(v, str), "non-string key " + str(v))
             self.solver.setOption(k, v)
 
+    def statistics(self):
+        """Return the statistics of this solver.
+
+        >>> c = Context()
+        >>> s = Solver(ctx=c)
+        >>> a = Int('a', c)
+        >>> s.add(a == 0)
+        >>> s.check()
+        sat
+        >>> stats = s.statistics()
+        >>> stats['cvc5::CONSTANT']
+        {'defaulted': False, 'internal': False, 'value': {'integer type': 1}}
+        >>> len(stats.get()) < 10
+        True
+        >>> len(stats.get(True, True)) > 30
+        True
```

Comment:
I'm not sure I understand what this should actually check?

Reply:
Hmm, but this is only the docstring?

Reply:
The docstrings are automatically analyzed for stuff that looks like a terminal session which is then used as unit tests (-> doctest)

Reply:
Fancy.